Code/Engine/Foundation/Basics/Platform/Win/HResultUtils.cpp

```diff
@@ -15,8 +15,8 @@ EZ_FOUNDATION_DLL ezString ezHRESULTtoString(ezMinWindows::HRESULT result)
 
   // Com error tends to put /r/n at the end. Remove it.
   ezStringBuilder message(ezStringUtf8(messageW).GetData());
-  message.ReplaceAll("\n", "");
-  message.ReplaceAll("\r", "");
+  message.ReplaceAll("\n"_ezsv, ""_ezsv);
```

Comment:
I just want to make this clear:
Because we moved ReplaceAll (and other stirng methods) form const char* to ezStringView. The generated code is not absolutely horrible.
`message.ReplaceAll("\n", "")` is equivalent to `message.ReplaceAll(ezStringView("\n"), ezStringView("")` which does a string length computation for each of the parameters. If you look at the disassembly it looks horrible.

By using _ezsv all this string length computation goes away, and the it esentially boils down to pointer + length which are both known at compile time.

Reply:
IIRC you mentioned checking the assembly with compiler explorer. Do you have a link to try this out?
Do you know whether there are differences in code generation between VS19 and VS20 ?
Would adding an always-inline to the constructor make a difference? Can we wrap it with pragmas to enforce optimization?

If I understood this correctly, the bad code generation mostly happens in Debug builds with MSVC, right?

If the compiler generates awful code, I would consider this a compiler bug. There is certainly some stuff that I'd be willing to do to guarantee best performance, but littering the entire code base with something that is pretty just telling the compiler to please not be so bad at its job, is going a bit far for my taste. So I wonder whether there are options that we have missed.